cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -273,16 +304,50 @@ def handle(name, cfg, cloud, log, args):
         raise RuntimeError(msg)
 
     config = ua_section.get("config")
-
     if config is not None:
         supplemental_schema_validation(config)
 
     maybe_install_ua_tools(cloud)
-    configure_ua(
-        token=ua_section.get("token"),
-        enable=ua_section.get("enable"),
-        config=config,
-    )
+
+    # ua-auto-attach.service had noop-ed as ua_section is not empty
+    features: dict = ua_section.get("features", {})
+    disable_auto_attach = bool(features.get("disable_auto_attach", False))
+    is_pro_cfg = not disable_auto_attach
+
+    is_pro = None
+    if is_pro_cfg:
+        try:
+            auto_attach_short(features)
+        except NotAProInstance:
+            is_pro = False
+        except RetryError:
+            is_pro = True
```

Comment:
If http_proxy is setup invalid on a system: is it possible we would hit a RetryError even if we are on a non-pro environment (because we never got ahold of the contracts.canonical.com to confirm whether PRO or not)?

If this is a possibility, I wonder if we really should be assuming `is_pro = True` just on a retry failure.

Reply:
After https://github.com/canonical/cloud-init/pull/1583/commits/6154d4d6a3ebe680f5111d69ea3a7a77598b8077, we are first check if we are in a Pro instance and if so we pass the http proxy config in https://github.com/canonical/cloud-init/pull/1583/files#diff-e12ed348cca4b27232907df295029b93b250cec08d438f0ab37e498a6df3a263R318

I guess `with_short_retry` will handle those possible errors, and we should just log any error that they raise.

Or do you, @CalvoM, think we should extra error handling?

Reply:
There is more changes we are making to the function for short retrial. I will fully update you, after some changes. e.g if we need to check if we are on a pro instance or we just use `should_auto_attach` function.